test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/Package/Download/PackageDownloadRunnerTests.cs

```diff
@@ -583,9 +581,6 @@ public async Task RunAsync_WithSourceMapping(
             settings,
             CancellationToken.None);
 
-        serverA.Stop();
-        serverB.Stop();
```

Comment:
This looks like a mistake. Since serverA isn't removed, it should be stopped.

Also, the code between the server.Start() and server.Stop() should be in a try-finally block, so the server.Stop() happens even if the other code throws an exception.